### PR TITLE
update astro starlight & blog plugin

### LINKED ISSUE #333: update astro starlight app (docs)
newest release is 0.29.2 and we're on 0.22.2

### PR DESCRIPTION
**Pull request by vincanger**
## Description

Fixes #333 

## Contributor Checklist

> Make sure to do the following steps if they are applicable to your PR:

- [ ] **Update e2e tests**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/template/e2e-tests](/template/e2e-tests) also.
- [x] **Update demo app**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/opensaas-sh/app_diff](/opensaas-sh/app_diff) also. Check [/opensaas-sh/README.md](/opensaas-sh/README.md) for details.
- [ ] **Update docs**: If needed, update the [/opensaas-sh/blog/src/content/docs](/opensaas-sh/blog/src/content/docs).

### CODE REVIEW

File: template/blog/src/content/docs/blog/2023-11-21-coverlettergpt.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 title: How I Built & Grew CoverLetterGPT to 5,000 Users and $200 MRR
 date: 2023-11-21
 tags: ["indiehacker", "saas", "sideproject"] 
+cover:
+  image: ../../../../public/og-images/2023-11-21-coverlettergpt.webp
```
Comment (Martinsos) [inline template/blog/src/content/docs/blog/2023-11-21-coverlettergpt.md:6]: Woudl be great if we didn't have to do this `../../../../` but I guess there is no way to somehow "alias" that path?

**Review by Martinsos — APPROVED** [review]
LGTM! Left one tiny comment.

**Comment by vincanger** [comment]
@Martinsos 

I just made lots of changes to this PR:

- I realized I was incorrectly setting og:images for documentation guides too, so created a default fallback. Nice thing is that now docs guides have og:images too
- Renamed `og-images` dir to `banner-images` since the blogposts automatically use images present in this folder as cover images as well, unless `hideBannerImage: false` is set on frontmatter
- Followed Astro's recommendations to use their Image component and certain settings to improve image perf. So now all docs/blogs images are in `src/assets` and use astro's Image component
- some code cleanup

File: opensaas-sh/blog/src/components/TitleWithBannerImage.astro [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    display: block;
+  }
+
+	h1 {
+		margin: 1rem 0;
+		font-size: var(--sl-text-h1);
+		line-height: var(--sl-line-height-headings);
+		font-weight: 600;
+		color: var(--sl-color-white);
+	}
```
Comment (Martinsos) [inline opensaas-sh/blog/src/components/TitleWithBannerImage.astro:44]: Woah, not keeping tabs on the tabs here :D!

**Review by Martinsos — APPROVED** [review]
That all sounds good to me @vincanger ! Let me konw if there is anything sepcific you want to me to check, if not, all good on my side (approved already).

File: opensaas-sh/blog/src/components/TitleWithBannerImage.astro (hunk 2 above)
Comment (vincanger) [reply opensaas-sh/blog/src/components/TitleWithBannerImage.astro:44]: yeah what the hell happened there?

fixed!

**Pull request merged**